app/configurator/interactive-filters/interactive-filters-config-options.tsx

```diff
@@ -216,6 +243,90 @@ const InteractiveTimeRangeFilterOptions = ({
   }
 };
 
+const InteractiveTimeSliderFilterOptions = ({
+  state: { chartConfig, dataSet, dataSource },
+}: {
+  state: ConfiguratorStateDescribingChart;
+}) => {
+  const locale = useLocale();
+  const [{ data }] = useDataCubeMetadataWithComponentValuesQuery({
+    variables: {
+      iri: dataSet,
+      sourceType: dataSource.type,
+      sourceUrl: dataSource.url,
+      locale,
+    },
+  });
+
+  const value =
+    get(chartConfig, "interactiveFiltersConfig.timeSlider.componentIri") ||
+    FIELD_VALUE_NONE;
+
+  if (data?.dataCubeByIri) {
+    const { dimensions, measures } = data.dataCubeByIri;
+    const allComponents = [...dimensions, ...measures];
+
+    const xComponentIri = getFieldComponentIri(chartConfig.fields, "x");
+    const xComponent = allComponents.find((d) => d.iri === xComponentIri);
+
+    const timeSliderDimensions = getTimeSliderFilterDimensions({
+      allComponents,
+      excludedComponent: isTemporalDimension(xComponent)
+        ? xComponent
+        : undefined,
+    });
```

Comment:
Nit: I think the getTimeSliderFilterDimensions helper should directly take the dataCubeByIri and the chart config. I think this way we do not have to "invent" a new API in the form of "allComponents" and "excludedComponent". This way it would also be easier to have this helper run directly on the redux + urql client cache if needed.

Reply:
Good point! I'll change that :)